homeassistant/components/timer.py

```diff
@@ -0,0 +1,261 @@
+def async_pause(hass, entity_id):
+    """Pause a timer."""
+    hass.async_add_job(hass.services.async_call(
+        DOMAIN, SERVICE_PAUSE, {ATTR_ENTITY_ID: entity_id}))
+
+
+@bind_hass
+def reset(hass, entity_id, duration):
+    """Reset a timer."""
+    hass.add_job(async_reset, hass, entity_id, duration)
+
+
+@callback
+@bind_hass
+def async_reset(hass, entity_id, duration):
+    """Reset a timer."""
+    hass.async_add_job(hass.services.async_call(
+        DOMAIN, SERVICE_RESET, {ATTR_ENTITY_ID: entity_id,
+                                ATTR_DURATION: duration}))
+
+
+@asyncio.coroutine
+def async_setup(hass, config):
+    """Set up a timer."""
+    component = EntityComponent(_LOGGER, DOMAIN, hass)
+
+    entities = []
+
+    for object_id, cfg in config[DOMAIN].items():
+        if not cfg:
+            cfg = {}
+
+        name = cfg.get(CONF_NAME)
+        initial = cfg.get(CONF_INITIAL)
+        icon = cfg.get(CONF_ICON)
+
+        entities.append(Timer(hass, object_id, name, initial, icon))
+
+    if not entities:
+        return False
+
+    @asyncio.coroutine
+    def async_handler_service(service):
+        """Handle a call to the timer services."""
+        target_timers = component.async_extract_from_service(service)
+
+        attr = None
+        if service.service == SERVICE_START:
+            attr = 'async_start'
+        elif service.service == SERVICE_PAUSE:
+            attr = 'async_pause'
+
+        tasks = [getattr(timer, attr)() for timer in target_timers if attr]
+        if service.service == SERVICE_RESET:
+            for timer in target_timers:
+                tasks.append(
+                    timer.async_reset(service.data.get(ATTR_DURATION,
+                                                       DEFAULT_INITIAL)))
+        if tasks:
+            yield from asyncio.wait(tasks, loop=hass.loop)
+
+    descriptions = yield from hass.async_add_job(
+        load_yaml_config_file, os.path.join(
+            os.path.dirname(__file__), 'services.yaml')
+    )
+
+    hass.services.async_register(
+        DOMAIN, SERVICE_START, async_handler_service,
+        descriptions[DOMAIN][SERVICE_START], SERVICE_SCHEMA)
+    hass.services.async_register(
+        DOMAIN, SERVICE_PAUSE, async_handler_service,
+        descriptions[DOMAIN][SERVICE_PAUSE], SERVICE_SCHEMA)
+    hass.services.async_register(
+        DOMAIN, SERVICE_RESET, async_handler_service,
+        descriptions[DOMAIN][SERVICE_RESET], SERVICE_SCHEMA_RESET)
+
+    yield from component.async_add_entities(entities)
+    return True
+
+
+class Timer(Entity):
+    """Representation of a timer."""
+
+    def __init__(self, hass, object_id, name, initial, icon):
+        """Initialize a timer."""
+        self.entity_id = ENTITY_ID_FORMAT.format(object_id)
+        self._name = name
+        self._state = self._initial = initial
+        self._icon = icon
+        self._status = STATUS_IDLE
+        self._hass = hass
+
+        async_track_time_interval(
```

Comment:
Don't track the interval, use `self.hass.helper.event.async_track_point_in_utc_time` on start of timer